src/common/transformations/src/transformations/common_optimizations/align_eltwise_input_ranks.cpp

```diff
@@ -0,0 +1,69 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/common_optimizations/align_eltwise_input_ranks.hpp"
+
+#include <ngraph/opsets/opset8.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+#include <ngraph/pattern/op/or.hpp>
+
+NGRAPH_RTTI_DEFINITION(ngraph::pass::AlignEltwiseInputRanks, "AlignEltwiseInputRanks", 0);
+
+ngraph::pass::AlignEltwiseInputRanks::AlignEltwiseInputRanks() {
+    auto input_pattern = pattern::any_input();
+    auto const_pattern = pattern::wrap_type<opset8::Constant>();
+    auto eltwise_pattern = pattern::wrap_type<opset8::SquaredDifference,
+                                              op::util::BinaryElementwiseComparison,
+                                              op::util::BinaryElementwiseLogical,
+                                              op::util::BinaryElementwiseArithmetic>({input_pattern, const_pattern}, pattern::has_static_rank());
+
+    auto input_low_pattern = pattern::wrap_type<opset8::Constant>();
+    auto input_high_pattern = pattern::wrap_type<opset8::Constant>();
+    auto output_low_pattern = pattern::wrap_type<opset8::Constant>();
+    auto output_high_pattern = pattern::wrap_type<opset8::Constant>();
+    auto fq_pattern = pattern::wrap_type<opset8::FakeQuantize>({input_pattern, input_low_pattern, input_high_pattern,
+                                                                output_low_pattern, output_high_pattern}, pattern::has_static_rank());
```

Comment:
I would put this operation types under single wrap_type, and then later in callback align all inputs that are Constants.